File: go/maximum-nesting-depth-of-the-parentheses.go
func maxDepth(s string) int {
	currentDepth := 0
	maximalDepth := 0
	for _, c := range s {
		if c == '(' {
			currentDepth += 1
		}
		if c == ')' {
			if maximalDepth < currentDepth {
				maximalDepth = currentDepth
			}
			currentDepth -= 1
		}
	}
	return maximalDepth
}

File: python/maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        current_depth = 0
        maximal_depth = 0
        for c in s:
            if c == "(":
                current_depth += 1
            elif c == ")":
                maximal_depth = max(maximal_depth, current_depth)
                current_depth -= 1
        return maximal_depth
